riff-raff/public/docs/riffraff/administration/auth.md

```diff
@@ -9,18 +9,20 @@ Authentication is handled by Google. This obtains the users full name and e-mail
-- Application layer: this layer uses The Guardian's [Play Google Auth Module](https://github.com/guardian/play-googleauth) and is configured in the main [application.conf](https://github.com/guardian/riff-raff/blob/afb7e602e11acd7a07aae433c74be22976d8a7cd/riff-raff/conf/application.conf#L40-L41).
-
-To rotate the secrets, follow these steps:
-1. Send a message about this rotation in the DevX Chat channel; this encourages people to report any unexpected problems to us quickly.
-1. Create a new Client Secret in Riff Raff's Google Cloud project.
-1. Patch the PROD and CODE application.conf to use the new secret.
-1. Update the `/INFRA/deploy/riff-raff/clientSecret` secret in AWS Secrets Manager.
+- Load Balancer layer: this component is defined in [the main riff raff cloudformation template](https://github.com/guardian/deploy-tools-platform/blob/main/cloudformation/riffraff/riffraff.template.yaml) and uses [authenticate-oidc](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html#configure-user-authentication). It relies on a secret stored in SSM.
+- Application layer: this layer uses The Guardian's [Play Google Auth Module](https://github.com/guardian/play-googleauth) and relies on secrets and configuration defined in the main [application.conf](https://github.com/guardian/riff-raff/blob/afb7e602e11acd7a07aae433c74be22976d8a7cd/riff-raff/conf/application.conf#L40-L41) stored in S3.
+
+To rotate these secrets we recommend a workflow that minimises risk by making it easy to perform a rollback until the very last step. At a high level, it consists of creating a new secret while keeping the old one active, switching the application to the new one, and only then disabling and deleting the old secret.
+
+Here is a step-by-step guide, which we recommend following first for the CODE environment and then on PROD:
+1. (PROD only) Send a message about this rotation in the DevX Chat channel; this encourages people to report any unexpected problems to us quickly.
+1. Create a new Client Secret in [Riff Raff's Google Cloud project](https://console.cloud.google.com/auth/clients?project=guardian-riff-raff) and leave the old secret in place.
+1. Patch the application.conf file stored in the deploy tools config S3 bucket to use the new secret.
+1. Update `/${STAGE}/deploy/riff-raff/clientSecret` in AWS Secrets Manager. As Secrets Manager is versioned, the new secret will be available as the most recent version with a given id.
 1. Update the [secret version id used in the CloudFormation template](https://github.com/guardian/deploy-tools-platform/blob/11730ea2841926148e98ab45e6d118bd1a133d27/cloudformation/riffraff/riffraff.template.yaml#L592). Merging your PR will trigger a [deployment to update the infrastructure](https://riffraff.gutools.co.uk/deployment/history?projectName=tools%3A%3Ariffraff-cloudformation&stage=PROD&pageSize=20&page=1).
```

Comment:
I think we need to update the link here?

```suggestion
1. Update the [secret version id for the relevant stage](https://github.com/guardian/deploy-tools-platform/blob/04591aa388c08c66a71461189c33add0b0cd4aa9/cloudformation/riffraff/riff-raff.yaml#L15-L19). Merging your PR will trigger a [deployment to update the infrastructure](https://riffraff.gutools.co.uk/deployment/history?projectName=tools%3A%3Ariffraff-cloudformation&stage=PROD&pageSize=20&page=1).
```

Reply:
Great catch thank you!